[WORKFLOWS] Fix deploy framework

0xdavinchee opened the pull request:
## Summary
- This PR fixes several ops-scripts for the new FlowNFT addition, namely the `deploy-framework.js`, `etherscan-verify.sh`, `info-print-contract-addresses.js`, `gov-upgrade-super-token-logic.js` scripts
- It also involves some cleanup in some other script files (the documentation) as well as some updates to the contract interface files

0xdavinchee committed:
deploy and link nft deployer if non-existent

0xdavinchee committed:
proper fix

0xdavinchee committed:
existence check in separate try/catch

0xdavinchee committed:
fix bad ordering

0xdavinchee committed:
Create force-gov-upgrade-super-token-logic.js

0xdavinchee committed:
temp force

0xdavinchee committed:
skip

0xdavinchee committed:
Update gov-upgrade-super-token-logic.js

0xdavinchee committed:
deploy script cleanup

- add verification for nft contracts
- fix supertoken verification (no need force args)
- add verification for nft deployer library

0xdavinchee committed:
test

0xdavinchee committed:
ops-scripts adjustments

0xdavinchee committed:
Merge branch 'dev' into fix-deploy-framework

d10r committed:
Merge branch 'fix-deploy-framework' into 1.6.0_deployment

d10r committed:
adapted verification

d10r committed:
fix jsdoc type

d10r committed:
token update script: support for batching, upgrade for 1.6.0 (FlowNFT creation), exclusion list

d10r committed:
undo var rename

d10r committed:
merge dev

d10r committed:
appease nazis

d10r committed:
ditto

d10r committed:
fix deploy-framework for from-scratch deployments

d10r committed:
fix visibility

hellwolf commented on packages/ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol:
```diff
@@ -45,6 +45,11 @@ interface ISuperToken is ISuperfluidToken, TokenInfo, IERC20, IERC777 {
+    /**************************************************************************
+    * Constants
+    *************************************************************************/
+    function SUPERFLUID_NFT_DEPLOYER_LIBRARY_ADDRESS() external view returns (address);
```
This should not be there, this is a detail that should not be part of the interface.

hellwolf replied:
The one uses this is using "SuperToken" contract, so it's not a problem for the use sites.

0xdavinchee replied:
will remove

hellwolf commented on packages/ethereum-contracts/contracts/interfaces/superfluid/ISuperTokenFactory.sol:
```diff
@@ -24,6 +24,16 @@ interface ISuperTokenFactory {
+     * @notice Get the canonical super token logic.
+     * @dev Is the same as `getSuperTokenLogic`
+     */
+    function _SUPER_TOKEN_LOGIC() external view returns(ISuperToken superTokenLogic);
```
external & _SUPER_TOKEN_LOGIC? (the underscore)

if it's the same as `getSuperTokenLogic`, why?

0xdavinchee replied:
fair, will also remove this one from interface

hellwolf commented on packages/ethereum-contracts/ops-scripts/gov-upgrade-super-token-logic.js:
```diff
@@ -1,8 +1,9 @@
 const {ZERO_ADDRESS} = require("./libs/common");
 
-const MAX_REQUESTS = 100;
+const MAX_REQUESTS = 20;
```
why the change?

d10r replied:
because a high number increases the chance of failed RPC requests, downside with minimal upside

hellwolf replied:
makes sense

0xdavinchee committed:
remove interface stuff

- change info-print script to use SuperToken artifact instead of ISuperToken because we are removing `SUPERFLUID_NFT_DEPLOYER_LIBRARY_ADDRESS` from the interface

hellwolf commented on packages/ethereum-contracts/contracts/interfaces/superfluid/ISuperTokenFactory.sol:
```diff
@@ -33,7 +37,8 @@ interface ISuperTokenFactory {
     /**
-     * @dev Get the current super token logic used by the factory
+     * @notice Get the canonical super token logic.
+     * @dev Is the same as `_SUPER_TOKEN_LOGIC`
```
remove

```
* @dev Is the same as `_SUPER_TOKEN_LOGIC`
```

0xdavinchee replied:
addressed

0xdavinchee replied:
addressed

hellwolf reviewed (changes requested):
"disabled-test:deployment:scripts-js-truffle": "yarn run-truffle test test/ops-scripts/deployment.test.js",

Should this still be disabled?

0xdavinchee commented:
> ```
>     "disabled-test:deployment:scripts-js-truffle": "yarn run-truffle test test/ops-scripts/deployment.test.js",
> ```
> 
> Should this still be disabled?

I don't think so, not sure why this is disabled since we use truffle to deploy not HH.

hellwolf commented:
> > ```
> >     "disabled-test:deployment:scripts-js-truffle": "yarn run-truffle test test/ops-scripts/deployment.test.js",
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Should this still be disabled?
> 
> I don't think so, not sure why this is disabled since we use truffle to deploy not HH.

It was disabled because it was broken. check the git blame. If it works again, let's uncomment it.

0xdavinchee committed:
address review comments

0xdavinchee committed:
Merge branch 'dev' into fix-deploy-framework

0xdavinchee committed:
redisable
